client/lib/signup/step-actions/index.js

```diff
@@ -1315,20 +1315,10 @@ export const buildUpgradeFunction = ( planProps, cartItem ) => {
 	};
 
 	if ( selectedSite && flowName === 'site-selected' && ! cartItem ) {
-		wpcom.req.post(
-			`/domains/${ selectedSite.ID }/${ selectedSite.name }/convert-domain-only-to-site`,
-			{},
-			( error ) => {
-				if ( error ) {
-					errorNotice( error.message );
-					return;
-				}
```

Comment:
Reverted D117774-code, I think we'll still need this on the front end (at least for now)

Reply:
Part of why I removed it is, at least locally, `selectedSite` would get cleared out I think with the automatic cache flushing that we do in dev. If we can figure out how to persist selectedSite via siteId and siteSlug query params, I think this could solve that issue. 

I think that that's somewhere over in [here](https://github.com/Automattic/wp-calypso/blob/trunk/client/signup/controller.js#L325). But, any time I tried to modify it, selectedSite would be empty when I clicked the free plan button.

Reply:
The removal made sense, just wanted to point out we'll need it until there is a backend fix in place.

> If we can figure out how to persist selectedSite via siteId and siteSlug query params, I think this could solve that issue.

Should be able to pass these through as step dependencies but I think you run into issues crossing between flows (like we might be?). There was a recent PR I happened to see that pushed siteId and slug into query params across their flow, if I'm understanding all this properly we could do the same: https://github.com/Automattic/wp-calypso/pull/79869/